src/app/contact/contact.component.ts

```diff
@@ -0,0 +1,42 @@
+
+import { Component, ViewChild, OnInit, ElementRef } from '@angular/core';
+import { ModalComponent } from 'ng2-bs3-modal/ng2-bs3-modal';
+@Component({
+    selector: 'app-contact',
+    templateUrl: './contact.component.html',
+    styleUrls: ['./contact.component.css']
+})
+export class ContactComponent implements OnInit {
+    @ViewChild('submitButton') submitButton;
+    @ViewChild('emailInput') emailInput;
+    nameInput: string;
+    //emailInput: string;
+    tpnoInput: number;
+    contactMessage: string = '';
+    constructor() { }
+
+    ngOnInit() {
+
+    }
+    @ViewChild('myModal')
+    modal: ModalComponent;
+
+    close() {
+        this.modal.close();
+    }
+
+    open() {
+        this.modal.open();
+    }
+    // check whether messsage contains morthan 100 words
+    checkWordCount() {
+
+        if (this.contactMessage.split(' ').length >= 100 && this.tpnoInput.toString().length>=10) {
```

Comment:
Word Count? We really, don't need much of these things right now. :sweat_smile: You can have @mariobehling opinion.